src/Features/ERDDAP/Superlatives/index.tsx

```diff
@@ -56,43 +63,81 @@ export const ShowSuperlatives: React.FunctionComponent<ShowSuperlativesProps> =
   unitSystem,
   laterThan,
 }) => {
-  const { platform: windPlatform, timeSeries: windTimeSeries } = findHighestCondition(platforms, laterThan, windSpeed)
-  const { platform: wavePlatform, timeSeries: waveTimeSeries } = findHighestCondition(platforms, laterThan, waveHeight)
-
-  if (windPlatform && windTimeSeries && wavePlatform && waveTimeSeries) {
-    return (
-      <Card style={{ marginTop: "1rem", marginBottom: "1rem" }}>
-        <CardHeader>
-          <h5>Latest Conditions</h5>
-        </CardHeader>
-
-        <CardBody>
-          <Row>
-            <Col>
+  const backOffHours = 6
+  const [windSuperlative, setWindSuperlative] = useState<HighestCondition>()
+  const [waveSuperlative, setWaveSuperlative] = useState<HighestCondition>()
+  const waveBackOffTime = useRef(laterThan)
+  const windBackOffTime = useRef(laterThan)
+
+  //Set wind superlative
+  useEffect(() => {
+    for (let hours = 0; hours < backOffHours; hours++) {
+      windBackOffTime.current = calcAnyHourAgoRounded(hours)
+      const { platform: windPlatform, timeSeries: windTimeSeries } = findHighestCondition(
+        platforms,
+        windBackOffTime.current,
+        windSpeed,
+      )
+      if (windPlatform || windTimeSeries) {
+        setWindSuperlative({ platform: windPlatform, timeSeries: windTimeSeries })
+        return
+      }
+    }
+  }, [platforms, laterThan])
```

Comment:
If I'm reading this right, this isn't starting with the `Date` passed in as `laterThan` but still generating a new `Date.now()` in the loop. 

It might be easiest to see this by running the Storybook tests with `yarn storybook` and then http://localhost:6006/?path=/docs/erddap-superlatives--docs will render https://github.com/gulfofmaine/Neracoos-1-Buoy-App/blob/main/src/Features/ERDDAP/Superlatives/index.stories.tsx

In the test I specify two different `laterThan`s to show the difference in how the component will render if it can find readings or not. 

https://github.com/gulfofmaine/Neracoos-1-Buoy-App/blob/76bec6d6faa6e2cba4d6ed9c906379c1760f44f1/src/Features/ERDDAP/Superlatives/index.stories.tsx#L19-L35

The first test should be able to show both of the conditions, and the second should miss both (it would be great to find a time/tweak things to demonstrate what happens if only one is found).

It's pulling the platform test data in from https://github.com/gulfofmaine/Neracoos-1-Buoy-App/blob/main/src/stories/platforms.ts

Reply:
Thanks for writing all that out. I misunderstood how the test was running.

Just so I don't make another incorrect assumption, would you mind confirming that I am thinking about the issue correctly?

- Because `<ShowSuperlative/>` wasn't finding a superlative in the `laterThan` timeframe, we are wanting to incrementally back up that time until we find a reading that can be a superlative. 
- We don't want this loop logic in the `<ShowSuperlatives />` component for testing's sake because we want to pass in a stubbed time to `<ShowSuperlatives/>` that reflects the stubbed data that will give us the above happy (superlatives found) and sad (no superlatives found) paths.

Reply:
I think we can have the backoff logic in `<ShowSuperlatives/>`, it just shouldn't have the time (the `Date.now()` call) that we are starting our search with in it, so that there is at least some separation between state and display. 

State in this case is the current selection of `platforms` and `laterThan` (`Date.now()`) that starts the search. In theory by splitting the state and the display, we can make sure that if we always pass the same inputs into `<ShowSuperlatives/>` then we will get the same outputs.

Reply:
Yes, definitely agree about the separation of state and display. 

I'm not following how its possible to do the superlatives logic in this component with the testing constraints you've laid out. To get the superlatives we have to pass a date as an arg into the  `findHighestCondition()` function and if we are looping through to back off the date, that date will have to change to determine whether superlatives exist in that timeframe.

I have a local branch going that refactors it so its parent component does the logic to 1) find the most recent time there are superlatives, and 2) passes just that date into the `<ShowSuperlatives/>` . This way the already predetermined date is being passed into the component and then doesn't change once its in the `<ShowSuperlatives/>` component.